app/main.py

```diff
@@ -1,2 +1,59 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 0.66) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.length = 0
+        self.limit = capacity * load_factor
+        self.hash_table = [[] for _ in range(capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length > self.limit:
+            self.resize()
+        index = self._get_index(key)
+        for i, (k, h, v) in enumerate(self.hash_table[index]):
+            if k == key and h == hash(key):
+                self.hash_table[index][i] = (key, hash(key), value)
+                return
```

Comment:
For you need this loop and check?)
You have this check in the _get_index method